File: src/main/kotlin/com/brandonswright/things/domain/game/Player.kt
package com.brandonswright.things.domain.game

class Player(val name: String, val id: String) {

    var role = PlayerRole.UNASSIGNED
        private set

    fun isReader() = role == PlayerRole.READER

    fun isWriter() = role == PlayerRole.WRITER

    fun makeReader() {
        role = PlayerRole.READER
    }

    fun makeWriter() {
        role = PlayerRole.WRITER
    }

    override fun toString(): String {
        return "Player(name='$name', id='$id', role=$role)"
    }
}

File: src/main/typescript/client/Player.ts
export class Player {
    public name: string
    public isReader: boolean
    public isWriter: boolean
    public id: string

    constructor(json: any) {
        this.name = json.name
        this.isReader = json.isReader
        this.isWriter = json.isWriter
        this.id = json.id
    }
}